agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/httpclient/RedirectPolicy.java

```diff
@@ -74,6 +84,9 @@
                 String responseLocation = httpResponse.getHeaderValue("Location");
                 if (responseLocation != null) {
                   redirectMappings.put(originalHttpRequest.getUrl(), responseLocation);
+                  if (instrumentationKey != null) {
+                    instrumentationKeyMappings.put(instrumentationKey, responseLocation);
+                  }
```

Comment:
I think should not populate `redirectMappings` if `instrumentationKey != null`?

(if/else similar to above when reading from the mappings)